config/env.config.js

```diff
@@ -55,5 +55,81 @@ const config = {
     }
 };
 
+const PLAYGROUND_ORIGIN = "playground_origin_environment_url";
+
+const PLAYGROUND_REDIRECT = "playground_redirect_environment_url";
+
+const RedirectPlugin = function(options) { 
+    this.apply = function(compiler) {
+        compiler.plugin('compilation', function(compilation) {
+            compilation.plugin('html-webpack-plugin-before-html-processing', function(htmlPluginData, callback) {
+                let htmlHead = htmlPluginData.html.match('<head>');
+                if (htmlHead) {
+                    htmlHead = htmlHead.index;
+                    htmlPluginData.html = htmlPluginData.html.slice(0, htmlHead) + '<head>' +
+                    `
+                    <script>
+                    (function() {
+                        function getParameterByName(name) {
+                            var url = window.location.search;
+                            var queryExp = new RegExp("[\\?&]"+name+"=([^&#]*)", "i");
+                            var match = queryExp.exec(url);
+                            if (match && match.length > 1) {
+                                return match[1];
+                            }
+                            return null;
+                        };
+                        var originUrl = (getParameterByName("originEnvironment") || "").toLowerCase();
+                        var targetUrl = (getParameterByName("targetEnvironment") || "").toLowerCase();
+
+                        // Set target environment for origin environment to redirect to
+                        if (targetUrl.length > 0) {
+                            targetUrl = decodeURIComponent(targetUrl)
+                            // Clear origin environment's local storage if target = origin
+                            if (window.location.href.toLowerCase().indexOf(targetUrl) != -1) {
+                                window.localStorage.removeItem("${PLAYGROUND_REDIRECT}");
+                                return;
+                            }
+
+                            window.localStorage.setItem("${PLAYGROUND_REDIRECT}", targetUrl);
+                        }
+
+                        // Redirect origin environment to target
+                        if (window.localStorage.getItem("${PLAYGROUND_REDIRECT}")) {
```

Comment:
We are making multiple calls to getItem with the same key. Better performance to store it in a local variable